azurerm/internal/services/recoveryservices/resource_arm_backup_policy_vm.go

```diff
@@ -313,6 +314,10 @@ func resourceArmBackupProtectionPolicyVMCreateUpdate(d *schema.ResourceData, met
 			return tf.ImportAsExistsError("azurerm_backup_policy_vm", *existing.ID)
 		}
 	}
+	// Less than 7 daily backups is no longer supported for create/update
+	if d.HasChange("retention_daily.0.count") && d.Get("retention_daily.0.count").(int) < 7 {
```

Comment:
where existing resources are grandfathered in (and there's been multiple years worth, so this isn't likely going to change anytime soon) - I think this wants to be:

```suggestion
	if (d.IsNewResource() || d.HasChange("retention_daily.0.count")) && d.Get("retention_daily.0.count").(int) < 7) {
```

which'll handle:

- new resources must have at least 7 days
- existing resources (with less than 7 days) can stay that way providing they're not updated